test/deploy/rackhd_stack_init.py

```diff
@@ -117,11 +117,13 @@ def test04_power_on_nodes(self):
             fit_common.countdown(30)
         # no PDU case
         else:
-            print '**** No supported PDU found, restarting nodes using IMPI.'
-            # Power off all nodes
-            self.assertNotEqual(fit_common.power_control_all_nodes("off"), 0, 'No BMC IP addresses found')
-            # Power on all nodes
-            self.assertNotEqual(fit_common.power_control_all_nodes("on"), 0, 'No BMC IP addresses found')
+            # Power cycle all nodes via IPMI, display warning if no nodes found
+            if fit_common.power_control_all_nodes("off") == 0:
+                print '**** No supported PDU found, restarting nodes using IMPI.'
+            else:
+                print '**** No BMC IP addresses found in arp table, continuing without node restart.'
```

Comment:
Can you double check if the error messages here are swapped?
It looks like power_control_all_nodes("off") returns 0 if no arp table entries, and non-zero if using ipmi to restart the nodes.